digid_eherkenning/models/base.py

```diff
@@ -166,7 +181,56 @@ class Meta:
     def __str__(self):
         return force_str(self._meta.verbose_name)
 
+    def populate_xml_fields(self, urls: dict[str, str], xml: str) -> None:
+        """
+        Populates the idp_metadata_file and idp_service_entity_id fields based on the
+        fetched xml metadata
+        """
+        self.idp_service_entity_id = urls["entityId"]
+        content = ContentFile(xml.encode("utf-8"))
+        self.idp_metadata_file.save("metadata.xml", content, save=False)
+
+    def process_metadata_from_xml_source(self) -> tuple[dict[str, str], str]:
+        """
+        Parses the xml metadata
+
+        :return a tuple of a dictionary with the useful urls and the xml string itself.
+        """
+        try:
+            xml = OneLogin_Saml2_IdPMetadataParser.get_metadata(
+                self.metadata_file_source
+            )
+            parsed_idp_metadata = OneLogin_Saml2_IdPMetadataParser.parse(
+                xml,
+                required_sso_binding=OneLogin_Saml2_Constants.BINDING_HTTP_POST,
+                required_slo_binding=OneLogin_Saml2_Constants.BINDING_HTTP_POST,
+            )
+        except Exception as exc:
```

Comment:
are there not more specific exceptions to catch? Now if you have a typo in `self.metata_file_source`, that will result in an `AttributeError` in our code being caught while it shouldn't.

Reply:
Basically these are handled by the `python3-saml` library (https://github.com/maykinmedia/python3-saml/blob/maykin/src/onelogin/saml2/idp_metadata_parser.py#L32). Is that correct?I mean do we need to explicitly raise an exception in `digid-eherkenning` library?

Reply:
of course they don't use proper exception classes to namespace things :( nothing we can do then, but please add a comment that the underlying library is raising bare exceptions